google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/sql/SqlType.java

```diff
@@ -210,6 +244,79 @@ static <K, V> SqlType.Map<K, V> mapOf(SqlType<K> keyType, SqlType<V> valType) {
     return mapOf(bytes(), arrayOf(struct()));
   }
 
+  /**
+   * Creates a fake {@link SqlType.Proto} for use on in {@link StructReader} methods that require a
+   * {@link SqlType} to validate against. This does not specify a schema because the proto schema
+   * will be validated on calls to the structs data accessors.
+   *
+   * <p>This is considered an internal implementation detail and not meant to be used by
+   * applications.
+   */
+  @InternalApi
+  static SqlType.Proto protoOf(com.google.bigtable.v2.Type.Proto protoType) {
+    return Type.SchemalessProto.fromProto(protoType);
```

Comment:
Why do we need the SchemalessProto variants here?

We have SchemalessStruct so that for e.g. `getMap("c", SqlType.mapOf(SqlType.string(), SqlType.struct())` the user doesn't have to specify the schema of the struct. We know that the struct schema will be validated on calls to the nested struct's getters.

I don't really understand how that maps to the proto usecase here. I would think we'd always want to return e.g Map<KeyType, ProtoMsgType> using Type.Proto and Type.Enum below

Reply:
Can we remove the SqlType factories for the schemaless varients and just directly construct Type.SchemalessProto, etc in fromProto instead? 

My main concern is that Im not sure users always notice the @InternalApi annotations and might end up using these incorrectly